clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -7548,6 +7546,17 @@ static void handleSYCLIntelMemoryAttr(Sema &S, Decl *D,
     if (MA->isImplicit())
       D->dropAttr<SYCLIntelMemoryAttr>();
 
+  // Check to see if there's a duplicate attribute with different values
+  // already applied to the declaration.
+  if (const auto *MA = D->getAttr<SYCLIntelMemoryAttr>()) {
```

Comment:
Is there a reason we're making these changes? I mean I don't see the harm but did we have a request to diagnose this? Just curious about what prompted this change. 

This check can also be combined with check above on L7545

Reply:
>>Is there a reason we're making these changes? I mean I don't see the harm but did we have a request to diagnose this? Just curious about what prompted this change.

@elizabethandrews, this part of https://github.com/intel/llvm/issues/11385

Reply:
>>This check can also be combined with check above on L7545

Sure, 
i will combine this

Reply:
>>This check can also be combined with check above on L7545

Done. Thanks for catching!